packages/wdio-utils/src/test-framework/testFnWrapper.js

```diff
@@ -0,0 +1,68 @@
+import { isFunctionAsync } from '../utils'
+import { logHookError } from './errorHandler'
+
+/**
+ * wraps test framework spec/hook function with WebdriverIO before/after hooks
+ *
+ * @param   {object} wrapFunctions  executeHooksWithArgs, executeAsync, runSync
+ * @param   {string} type           Test/Step or Hook
+ * @param   {object} spec           specFn and specFnArgs
+ * @param   {object} before         beforeFn and beforeFnArgs
+ * @param   {object} after          afterFn and afterFnArgs
+ * @param   {string} cid            cid
+ * @param   {number} repeatTest     number of retries if test fails
+ * @return  {*}                     specFn result
+ */
+export const testFrameworkFnWrapper = async function (
+    { executeHooksWithArgs, executeAsync, runSync },
+    type,
+    { specFn, specFnArgs },
```

Comment:
Should `specFnArgs` have default value as `beforeFnArgs` and `afterFnArgs`?

Reply:
I have set default values for after fn args because I'm modifying it.